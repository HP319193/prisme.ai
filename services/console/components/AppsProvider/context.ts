import { createContext, useContext } from 'react';

export interface AppsContext {
  apps: Map<string, Prismeai.App>;
  appInstances: Map<string, Prismeai.DetailedAppInstance[]>;
  getApps: ({
    text,
    page,
    limit,
    workspaceId,
  }: {
    text?: PrismeaiAPI.SearchApps.QueryParameters['text'];
    page?: PrismeaiAPI.SearchApps.QueryParameters['page'];
    limit?: PrismeaiAPI.SearchApps.QueryParameters['limit'];
    workspaceId?: PrismeaiAPI.SearchApps.QueryParameters['workspaceId'];
  }) => Promise<Prismeai.App[] | null>;
  getAppInstances: (
    workspaceId: PrismeaiAPI.ListAppInstances.PathParameters['workspaceId']
  ) => Promise<Prismeai.DetailedAppInstance[]>;
}

export const appsContext = createContext<AppsContext>({
  apps: new Map(),
  appInstances: new Map(),
  getApps: async () => ({} as any),
  getAppInstances: async () => ({} as any),
});

export const useApps = () => useContext(appsContext);

export default appsContext;
