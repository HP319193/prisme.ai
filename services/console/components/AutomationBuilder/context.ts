import { Schema } from '@prisme.ai/design-system';
import { createContext, useContext } from 'react';

export interface AutomationBuilderContext {
  automationId: string;
  addInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  removeInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  removeCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  editTrigger: () => void;
  editOutput: () => void;
  getApp: (
    instruction: string
  ) => {
    name: string;
    icon: string;
    instructionName: string;
  };
  instructionsSchemas: [
    string,
    Record<string, Schema & { description?: string }>,
    { icon: string }
  ][];
  getSchema: (name: string) => Schema;
}
export const automationBuilderContext = createContext<AutomationBuilderContext>(
  {
    automationId: '',
    addInstruction() {},
    removeInstruction() {},
    editInstruction() {},
    editCondition() {},
    removeCondition() {},
    editTrigger() {},
    editOutput() {},
    getApp() {
      return {
        name: '',
        icon: '',
        instructionName: '',
      };
    },
    instructionsSchemas: [],
    getSchema() {
      return {} as Schema;
    },
  }
);

export const useAutomationBuilder = () => useContext(automationBuilderContext);

export default automationBuilderContext;
