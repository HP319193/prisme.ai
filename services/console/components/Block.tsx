import * as React from 'react';
import * as prismeaiDS from '@prisme.ai/design-system';
import * as prismeaiSDK from '../utils/api';
import { ReactElement, useEffect, useState } from 'react';
import { nanoid } from 'nanoid';
import { useTranslation } from 'next-i18next';

if (process.browser) {
  // @ts-ignore
  window.__external = window.__external || {
    React: { ...React, default: React },
    prismeaiDS,
    prismeaiSDK,
  };
}

class BlockErrorBoundary extends React.Component {
  state = {
    hasError: false,
  };

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  componentDidCatch(error: any, errorInfo: any) {
    // Vous pouvez aussi enregistrer l'erreur au sein d'un service de rapport.
    console.error(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      // Vous pouvez afficher n'importe quelle UI de repli.
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;
  }
}

interface BlockComponentProps {
  entityId: string;
  token?: string;
  workspaceId?: string;
  appInstance?: string;
  language?: string;
}
type BlockComponent = (props: BlockComponentProps) => ReactElement;

interface BlockProps extends BlockComponentProps {
  url: string;
  renderLoading?: ReactElement;
}

export const ReactBlock = ({ url, renderLoading, ...props }: BlockProps) => {
  const {
    i18n: { language },
  } = useTranslation('workspaces');
  const [loading, setLoading] = useState(true);

  const [Component, setComponent] = useState<BlockComponent | null>(null);
  useEffect(() => {
    const uniqMethod = `__load_${nanoid()}`;
    // @ts-ignore
    window[uniqMethod] = (module) => {
      setComponent(() => {
        return module.default;
      });
      setLoading(false);
    };
    const s = document.createElement('script');

    s.innerText = `
    import * as module from '${url}';
    try {
      window['${uniqMethod}'](module);
    } catch (e) {}
    `;
    s.type = 'module';
    document.body.appendChild(s);

    return () => {
      // @ts-ignore
      delete window[uniqMethod];
    };
  }, [url]);

  return (
    <>
      {(loading && renderLoading) || null}
      {Component && <Component {...props} language={language} />}
    </>
  );
};

export const IFrameBlock = ({
  url,
  entityId,
  token,
  renderLoading,
}: BlockProps) => {
  const [loading, setLoading] = useState(true);
  const [height, setHeight] = useState(100);
  const handleLoad = React.useCallback(
    (e) => {
      setLoading(false);
      e.target.contentWindow.postMessage(
        {
          source: 'prisme.ai',
          token,
          id: entityId,
        },
        '*'
      );
    },
    [entityId, token]
  );
  useEffect(() => {
    const listener = (e: MessageEvent) => {
      if (e.data.source === entityId) {
        Object.keys(e.data).forEach((method) => {
          switch (method) {
            case 'init':
              setHeight(+e.data[method].height);
          }
        });
      }
    };
    window.addEventListener('message', listener);

    return () => {
      window.removeEventListener('message', listener);
    };
  }, [entityId, height]);

  return (
    <>
      {(loading && renderLoading) || null}
      <iframe
        src={url}
        onLoad={handleLoad}
        className="flex"
        style={{
          width: '100%',
          height: `${height}px`,
        }}
      />
    </>
  );
};

export const Block = ({ url, ...props }: BlockProps) => {
  const isJs = url.match(/\.js$/);
  if (isJs) {
    return (
      <BlockErrorBoundary>
        <ReactBlock url={url} {...props} />
      </BlockErrorBoundary>
    );
  }
  return <IFrameBlock url={url} {...props} />;
};

export default Block;
