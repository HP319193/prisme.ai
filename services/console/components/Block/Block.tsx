export const Block = () => {
  return <div>Block yeah</div>;
};

export default Block;
