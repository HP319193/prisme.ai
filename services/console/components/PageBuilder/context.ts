import { createContext } from 'react';
import { Schema } from '@prisme.ai/design-system';
import { BlockInCatalog } from './useBlocks';
import { useContext } from '../../utils/useContext';

export type Block = NonNullable<Prismeai.Page['blocks']>[number];
export type BlocksWithKeys = Map<string, Block>;

export interface PageBuilderContext {
  value: BlocksWithKeys;
  addBlock: (position: number, blockName?: string) => void;
  setEditBlock: (blockId: string) => void;
  removeBlock: (key: string) => void;
  setBlockConfig: (key: string, config: any) => void;
  setBlockSchema: (blockId: string, schema: Schema | null) => void;
  catalog: BlockInCatalog[];
  blocksSchemas: Map<string, Schema | null>;
}

export const context = createContext<PageBuilderContext | undefined>(undefined);

export const usePageBuilder = () => useContext<PageBuilderContext>(context);
