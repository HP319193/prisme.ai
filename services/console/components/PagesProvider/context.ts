import { createContext, Dispatch, SetStateAction, useContext } from 'react';

type WorkspaceId = string;

export interface PagesContext {
  pages: Map<WorkspaceId, Set<Prismeai.Page>>;
  fetchPages: (workspaceId: NonNullable<Prismeai.Workspace['id']>) => void;
  setPages: Dispatch<SetStateAction<Map<string, Set<Prismeai.Page>>>>;
  createPage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  savePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  deletePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    pid: NonNullable<Prismeai.Page['id']>
  ) => Promise<Pick<Prismeai.Page, 'id'>>;
}

export const pagesContext = createContext<PagesContext>({
  pages: new Map(),
  setPages() {},
  fetchPages() {},
  async createPage() {
    return {} as Prismeai.Page;
  },
  async savePage() {
    return {} as Prismeai.Page;
  },
  async deletePage() {
    return {} as Prismeai.Page;
  },
});

export const usePages = () => useContext(pagesContext);

export default usePages;
