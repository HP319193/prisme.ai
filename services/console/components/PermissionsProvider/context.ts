import { createContext, useContext } from 'react';
import UserPermissions = Prismeai.UserPermissions;
import SubjectType = PrismeaiAPI.GetPermissions.Parameters.SubjectType;

export interface PermissionsContext {
  usersPermissions: Map<string, UserPermissions[]>;
  addUserPermissions: (
    subjectType: SubjectType,
    subjectId: string,
    permissions: UserPermissions
  ) => Promise<any>;
  // unshareWorkspaceToUser: (
  //   userEmail: string,
  //   workspaceId: string
  // ) => Promise<any>;
  getUsersPermissions: (
    subjectType: SubjectType,
    subjectId: string
  ) => Promise<UserPermissions[]>;
}

export const workspacesContext = createContext<PermissionsContext>({
  usersPermissions: new Map(),
  // unshareWorkspaceToUser: async () => ({} as any),
  addUserPermissions: async () => ({} as UserPermissions),
  getUsersPermissions: async () => [] as UserPermissions[],
});

export const usePermissions = () => useContext(workspacesContext);

export default workspacesContext;
