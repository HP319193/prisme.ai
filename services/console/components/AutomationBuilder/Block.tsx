import Image from 'next/image';
import useHover from '@react-hook/hover';
import React, {
  FC,
  LegacyRef,
  memo,
  ReactNode,
  useCallback,
  useRef,
} from 'react';
import { NodeProps } from 'react-flow-renderer';
import { Flow } from './flow';
import { useAutomationBuilder } from './context';
import { Trans, useTranslation } from 'next-i18next';
import { truncate } from '../../utils/strings';
import { DeleteOutlined, EditOutlined } from '@ant-design/icons';

interface BlockProps {
  removable?: boolean;
  onEdit?: () => void;
  displayAs?: 'trigger' | 'instruction' | 'output' | 'condition' | 'repeat';
}

interface BlockClassName {
  textClassName: string;
  bgClassName: string;
  borderClassName: string;
  editTextClassName?: string;
}

interface BlockUI {
  editSection: ReactNode;
  topContent: ReactNode;
  selected: boolean;
  onEdit?: () => void;
  blockClassName: BlockClassName;
}

const CLASSNAME_BLOCKS: Record<string, BlockClassName> = {
  trigger: {
    textClassName: 'text-accent',
    bgClassName: 'bg-graph-background',
    borderClassName: 'border-accent',
  },
  instruction: {
    textClassName: '',
    bgClassName: 'bg-white',
    borderClassName: 'border-slate-300',
  },
  output: {
    textClassName: 'text-green-400',
    bgClassName: 'bg-green-200',
    borderClassName: 'border-green-400',
  },
  condition: {
    textClassName: '',
    editTextClassName: '!text-slate-400',
    bgClassName: 'bg-white',
    borderClassName: 'border-slate-300',
  },
};

// eslint-disable-next-line react/display-name
const BlockUI = React.forwardRef(
  (
    { blockClassName, editSection, topContent, selected, onEdit }: BlockUI,
    ref: LegacyRef<HTMLDivElement> | undefined
  ) => (
    <div
      className={`
          flex
          flex-col
          surface-section
          ${blockClassName.textClassName}
          ${blockClassName.bgClassName}
          ${blockClassName.borderClassName}
          border-[1px]
          ${selected ? 'drop-shadow-xl' : ''}
          transition-all ease
          rounded`}
      style={{ width: Flow.BLOCK_WIDTH - 50 }}
      ref={ref}
      onClick={onEdit}
    >
      <div
        className={`
          flex
          border-b-[1px]
          ${blockClassName.borderClassName}
          font-bold
          p-2
        `}
      >
        {topContent}
      </div>
      <button
        className={`
          flex
          p-2
          justify-between
          items-center
          ${blockClassName.editTextClassName || ''}
          `}
        style={{
          background: 'none',
          border: '0',
          fontSize: 'inherit',
          cursor: `pointer`,
        }}
      >
        {editSection}
        {onEdit ? <EditOutlined /> : null}
      </button>
    </div>
  )
);

export const Block: FC<NodeProps & BlockProps> = ({
  data,
  removable = true,
  selected,
  onEdit,
  displayAs,
}) => {
  const { t } = useTranslation('workspaces');
  const { removeInstruction, getApp } = useAutomationBuilder();
  const ref = useRef(null);
  const isHover = useHover(ref);
  const { icon, name } = getApp(data.label);

  const getLabel = useCallback(
    ({ label, value }: { label: string; value: any }) => {
      let displayedValue = '';
      switch (label) {
        case 'emit':
          displayedValue =
            typeof value === 'string' ? value : (value && value.event) || '';
          break;
        case 'repeat':
          if (!value) {
            displayedValue = '?';
          }
          if (value.on) {
            displayedValue = value.on;
          }
          if (value.until) {
            return t('automations.instruction.label', {
              context: 'repeat_until',
              count: +value.until,
            });
          }
          displayedValue = (value && value.on) || value.until || '?';
          break;
        case 'set':
          displayedValue =
            value &&
            `${value.name} = ${truncate(JSON.stringify(value.value), 10, 'â€¦')}`;
          break;
        case 'delete':
          displayedValue = (value && value.name) || '?';
          break;
        case 'wait':
          displayedValue = (value && value.event) || '?';
          break;
        case 'comment':
          return (
            <div className="italic text-neutral-500">
              {typeof value === 'string'
                ? value
                : JSON.stringify(value, null, '  ')}
            </div>
          );
        case 'output':
          return (
            <Trans
              t={t}
              i18nKey="automations.output.label"
              components={{
                pre: <pre className="text-left text-xs" />,
                code: <code />,
                italic: <div className="text-xs text-gray-50 italic" />,
              }}
              values={{
                context: value.output ? 'json' : 'empty',
                output: (
                  (typeof value.output === 'string'
                    ? value.output
                    : JSON.stringify(value.output, null, '  ')) || ''
                ).replaceAll('{{', '{\u200b{'),
              }}
            >
              output
            </Trans>
          );
        default:
          displayedValue =
            typeof value === 'string' ? value : (value && value.event) || '';
      }
      return t('automations.node.label', {
        instruction: t('automations.instruction.label', { context: label }),
        value: displayedValue,
        display: ':',
        interpolation: {
          skipOnVariables: true,
        },
      });
    },
    [t]
  );

  return (
    <BlockUI
      ref={ref}
      blockClassName={
        (displayAs && CLASSNAME_BLOCKS[displayAs]) ||
        CLASSNAME_BLOCKS['instruction']
      }
      topContent={
        <>
          {icon && (
            <div className="mr-2">
              <Image src={icon} width={16} height={16} alt={name} />
            </div>
          )}
          <div className="flex flex-1 justify-between">
            {data.title
              ? t('automations.node.title', { context: data.title })
              : name}
            {removable && (
              <button
                className="border-none cursor-pointer"
                style={{
                  background: 'none',
                  visibility: isHover ? 'visible' : 'hidden',
                }}
                onClick={() => removeInstruction(data.parent, data.index)}
              >
                <DeleteOutlined className="!text-orange-500" />
              </button>
            )}
          </div>
        </>
      }
      editSection={data.component ? <data.component /> : getLabel(data)}
      selected={selected}
      onEdit={onEdit}
    />
  );
};

export default memo(Block);
