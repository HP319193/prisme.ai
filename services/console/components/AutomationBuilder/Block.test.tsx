import Block from './Block';
import renderer, { act } from 'react-test-renderer';
import { useAutomationBuilder } from './context';

jest.mock('react-flow-renderer', () => {
  return {
    Handle: () => <div>Handle</div>,
    Position: {
      Bottom: 'bottom',
      Top: 'top',
    },
  };
});

jest.mock('./context', () => {
  const mock = {
    getApp: () => ({
      name: 'Logical',
      icon: '/icon.svg',
    }),
    removeInstruction: jest.fn(),
  };
  return {
    useAutomationBuilder: () => mock,
  };
});

it('should render', () => {
  const root = renderer.create(
    <>
      <Block
        id="a"
        data={{ label: 'emit', value: { event: 'the event' } }}
        type="instruction"
        selected={false}
        isConnectable={false}
      />
      );
      <Block
        id="a"
        data={{ label: 'repeat', value: { on: '$a', do: [] } }}
        type="instruction"
        selected={false}
        isConnectable={false}
      />
      );
      <Block
        id="a"
        data={{ label: 'set', value: { name: 'foo', value: 'bar' } }}
        type="instruction"
        selected={false}
        isConnectable={false}
      />
      );
      <Block
        id="a"
        data={{ label: 'delete', value: { name: 'foo' } }}
        type="instruction"
        selected={false}
        isConnectable={false}
      />
      );
      <Block
        id="a"
        data={{ label: 'wait' }}
        type="instruction"
        selected={false}
        isConnectable={false}
      />
      );
      <Block
        id="a"
        data={{ label: 'do something' }}
        type="instruction"
        selected={false}
        isConnectable={false}
      />
      );
    </>
  );
  expect(root.toJSON()).toMatchSnapshot();
});

it('should remove instruction', () => {
  const parent: any = [];
  const root = renderer.create(
    <Block
      removable
      id="a"
      data={{ parent, index: 0, label: 'emit', value: { event: 'the event' } }}
      type="instruction"
      selected={false}
      isConnectable={false}
    />
  );

  act(() => {
    root.root
      .find((el) => el.props.className === 'pi pi-times-circle')
      .parent!.props.onClick();
  });
  expect(useAutomationBuilder().removeInstruction).toHaveBeenCalledWith(
    parent,
    0
  );
});
