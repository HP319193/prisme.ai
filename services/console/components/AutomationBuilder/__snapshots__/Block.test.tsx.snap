// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Array [
  <div
    className="
          flex
          flex-col
          surface-section
          
          bg-white
          border-graph-border
          border-[1px]
          
          transition-all ease
          rounded"
    style={
      Object {
        "width": 260,
      }
    }
  >
    <div
      className="
          flex
          border-b-[1px]
          border-graph-border
          font-semibold
          p-[1.15rem]
          pb-3.5
        "
    >
      <div
        className="mr-3 flex justify-center items-center"
      >
        <img
          alt="Logical"
          className="object-contain h-[1rem] w-[1rem]"
          src="/icon.svg"
        />
      </div>
      <div
        className="flex flex-1 justify-between"
      >
        Logical
        <button
          className="border-none cursor-pointer flex justify-center items-center"
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "visibility": "hidden",
            }
          }
        />
      </div>
    </div>
    <button
      className="
          flex
          p-3.5
          justify-between
          items-center
          
          font-light
          overflow-hidden
          text-left
          "
      style={
        Object {
          "background": "none",
          "border": "0",
          "cursor": "pointer",
          "fontSize": "inherit",
        }
      }
    >
      automations.node.label
    </button>
  </div>,
  ");",
  <div
    className="
          flex
          flex-col
          surface-section
          
          bg-white
          border-graph-border
          border-[1px]
          
          transition-all ease
          rounded"
    style={
      Object {
        "width": 260,
      }
    }
  >
    <div
      className="
          flex
          border-b-[1px]
          border-graph-border
          font-semibold
          p-[1.15rem]
          pb-3.5
        "
    >
      <div
        className="mr-3 flex justify-center items-center"
      >
        <img
          alt="Logical"
          className="object-contain h-[1rem] w-[1rem]"
          src="/icon.svg"
        />
      </div>
      <div
        className="flex flex-1 justify-between"
      >
        Logical
        <button
          className="border-none cursor-pointer flex justify-center items-center"
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "visibility": "hidden",
            }
          }
        />
      </div>
    </div>
    <button
      className="
          flex
          p-3.5
          justify-between
          items-center
          
          font-light
          overflow-hidden
          text-left
          "
      style={
        Object {
          "background": "none",
          "border": "0",
          "cursor": "pointer",
          "fontSize": "inherit",
        }
      }
    >
      automations.node.label
    </button>
  </div>,
  ");",
  <div
    className="
          flex
          flex-col
          surface-section
          
          bg-white
          border-graph-border
          border-[1px]
          
          transition-all ease
          rounded"
    style={
      Object {
        "width": 260,
      }
    }
  >
    <div
      className="
          flex
          border-b-[1px]
          border-graph-border
          font-semibold
          p-[1.15rem]
          pb-3.5
        "
    >
      <div
        className="mr-3 flex justify-center items-center"
      >
        <img
          alt="Logical"
          className="object-contain h-[1rem] w-[1rem]"
          src="/icon.svg"
        />
      </div>
      <div
        className="flex flex-1 justify-between"
      >
        Logical
        <button
          className="border-none cursor-pointer flex justify-center items-center"
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "visibility": "hidden",
            }
          }
        />
      </div>
    </div>
    <button
      className="
          flex
          p-3.5
          justify-between
          items-center
          
          font-light
          overflow-hidden
          text-left
          "
      style={
        Object {
          "background": "none",
          "border": "0",
          "cursor": "pointer",
          "fontSize": "inherit",
        }
      }
    >
      automations.node.label
    </button>
  </div>,
  ");",
  <div
    className="
          flex
          flex-col
          surface-section
          
          bg-white
          border-graph-border
          border-[1px]
          
          transition-all ease
          rounded"
    style={
      Object {
        "width": 260,
      }
    }
  >
    <div
      className="
          flex
          border-b-[1px]
          border-graph-border
          font-semibold
          p-[1.15rem]
          pb-3.5
        "
    >
      <div
        className="mr-3 flex justify-center items-center"
      >
        <img
          alt="Logical"
          className="object-contain h-[1rem] w-[1rem]"
          src="/icon.svg"
        />
      </div>
      <div
        className="flex flex-1 justify-between"
      >
        Logical
        <button
          className="border-none cursor-pointer flex justify-center items-center"
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "visibility": "hidden",
            }
          }
        />
      </div>
    </div>
    <button
      className="
          flex
          p-3.5
          justify-between
          items-center
          
          font-light
          overflow-hidden
          text-left
          "
      style={
        Object {
          "background": "none",
          "border": "0",
          "cursor": "pointer",
          "fontSize": "inherit",
        }
      }
    >
      automations.node.label
    </button>
  </div>,
  ");",
  <div
    className="
          flex
          flex-col
          surface-section
          
          bg-white
          border-graph-border
          border-[1px]
          
          transition-all ease
          rounded"
    style={
      Object {
        "width": 260,
      }
    }
  >
    <div
      className="
          flex
          border-b-[1px]
          border-graph-border
          font-semibold
          p-[1.15rem]
          pb-3.5
        "
    >
      <div
        className="mr-3 flex justify-center items-center"
      >
        <img
          alt="Logical"
          className="object-contain h-[1rem] w-[1rem]"
          src="/icon.svg"
        />
      </div>
      <div
        className="flex flex-1 justify-between"
      >
        Logical
        <button
          className="border-none cursor-pointer flex justify-center items-center"
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "visibility": "hidden",
            }
          }
        />
      </div>
    </div>
    <button
      className="
          flex
          p-3.5
          justify-between
          items-center
          
          font-light
          overflow-hidden
          text-left
          "
      style={
        Object {
          "background": "none",
          "border": "0",
          "cursor": "pointer",
          "fontSize": "inherit",
        }
      }
    >
      automations.node.label
    </button>
  </div>,
  ");",
  <div
    className="
          flex
          flex-col
          surface-section
          
          bg-white
          border-graph-border
          border-[1px]
          
          transition-all ease
          rounded"
    style={
      Object {
        "width": 260,
      }
    }
  >
    <div
      className="
          flex
          border-b-[1px]
          border-graph-border
          font-semibold
          p-[1.15rem]
          pb-3.5
        "
    >
      <div
        className="mr-3 flex justify-center items-center"
      >
        <img
          alt="Logical"
          className="object-contain h-[1rem] w-[1rem]"
          src="/icon.svg"
        />
      </div>
      <div
        className="flex flex-1 justify-between"
      >
        Logical
        <button
          className="border-none cursor-pointer flex justify-center items-center"
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "visibility": "hidden",
            }
          }
        />
      </div>
    </div>
    <button
      className="
          flex
          p-3.5
          justify-between
          items-center
          
          font-light
          overflow-hidden
          text-left
          "
      style={
        Object {
          "background": "none",
          "border": "0",
          "cursor": "pointer",
          "fontSize": "inherit",
        }
      }
    >
      automations.node.label
    </button>
  </div>,
  ");",
]
`;
